src/utilities/alfalfa/AlfalfaPoint_Impl.hpp

```diff
@@ -17,10 +17,13 @@ namespace alfalfa {
       AlfalfaPoint_Impl();
 
      protected:
+      // TODO: is this pImpl is needed? If so, add getters and setters and make all members private.
```

Comment:
TODO

Reply:
It's annoying to write for sure. For ModelObjects we actually have ruby scripts that will do 90% of the work for us and generate the pimpl class: https://github.com/NREL/OpenStudio/blob/develop/developer/ruby/GenerateClass.rb and https://github.com/NREL/OpenStudio/blob/develop/developer/doc/Adding_Model_Object_to_OpenStudio.docx

I had written this when I first touched the OpenStudio codebase: https://github.com/jmarrec/OpenStudio-Tutorial-for-Cpp-Noobs?tab=readme-ov-file#pimpl

a few years later, I think the pImpl pattern in the case of model objects at least is useful for us because:

* The Impl methods are not exposed to SWIG (ruby, python, C# bindings)
* Technically you can change the Impl header without changing the public header, which is good because:
    * If a class never included the Impl header (and we have to do that too often for casting especially), then there's a net compilation time benefit
    * It forces us to treat the public header as a sacred API, and non-backward compatible changes must be really  motivated. We often remove methods from the Impl when they become moot, and we'll put a `[[deprecated]]` method in the public side for 2 or 3 releases with a clear message that it'll be removed.

Reply:
That makes sense. The reason I used the pImpl pattern here was to allow the point object a user is given after creation and the point object in the points vector within `AlflalfaJSON` to all point to the same underlying data. This seemed like the best option outside of dealing with just raw or lightly wrapped pointers. If there is a better paradigm you have in mind for this I would be happy to implement it.